src/server/mod.rs

```diff
@@ -419,7 +423,24 @@ where
                         Command::Pasv => {
                             ensure_authenticated!();
 
-                            let listener = std::net::TcpListener::bind(&passive_addrs.as_slice())?;
+                            let mut rng = rand::thread_rng();
+
+                            let mut listener: Option<std::net::TcpListener> = None;
+                            for _ in 1..BIND_RETRIES {
```

Comment:
Does the loop end when an open port is found?

Reply:
right... ofcourse :-S

Reply:
added break statement